crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__black_test__function_trailing_comma_py.snap

```diff
@@ -113,23 +113,24 @@ some_module.some_function(
 -def xxxxxxxxxxxxxxxxxxxxxxxxxxxx() -> (
 -    Set["xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"]
 -):
--    json = {
--        "k": {
--            "k2": {
--                "k3": [
--                    1,
--                ]
--            }
--        }
--    }
 +def xxxxxxxxxxxxxxxxxxxxxxxxxxxx() -> Set[
 +    "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
 +]:
-+    json = {"k": {"k2": {"k3": [1]}}}
+     json = {
+         "k": {
+             "k2": {
+                 "k3": [
+                     1,
```

Comment:
Interesting that black doesn't add trailing commas to all outer elements, especially considering that the trailing comma in the array expression isn't JSON.

We can tackle this divergence as its own PR if you like. It probably requires:
* Detecting that the dict (and list) only contains JSON like values
* Add a `Preserve` options to the `comma_separated` builder to only preserve existing trailing commas.